### PR TITLE
Better seeds for demo

### PR DESCRIPTION
**Pull request by kierachell**
connects #663 

- added `vbms_id` to `admin/establish_claim` to make sure it matches input
- cleaned a couple of tests
- added more certifications to do; so that form8s can be refilled with new data;
- fixed `uncertify` on `already certified` view to display properly
- killed Anne Merica

### CODE REVIEW

**Comment by kierachell** [comment]
tagged @joofsh because of `admin/establish_claim` change
tagged @mdbenjam because of the existing EP type seeds change

File: app/views/certifications/confirm.html.erb [hunk 1]
```diff
@@ -10,6 +10,9 @@
   </ul>
 
   <p class="cf-msg-screen-text">Way to go! You can now close this browser window and open another case in VACOLS.</p>
+  <% if Rails.env.development? || Rails.env.demo? || Rails.env.test? %>
```
Comment (shanear) [inline app/views/certifications/confirm.html.erb]: I think we should try and avoid test only UI unless its necessary. Is there a better way to do this?

File: lib/fakes/bgs_service.rb [hunk 2]
```diff
@@ -24,96 +24,21 @@ class Fakes::BGSService
...
-        benefit_claim_id: "17",
-        claim_receive_date: Time.zone.now - 10.days,
-        claim_type_code: "930RCPMC",
-        status_type_code: "PEND"
       }
     ].freeze
 
   def get_end_products(_veteran_id)
-    end_product_data || END_PRODUCTS
+    ep_data = [END_PRODUCTS[0], END_PRODUCTS[1], END_PRODUCTS[2]]
```
Comment (shanear) [inline lib/fakes/bgs_service.rb]: you'll need to work out this change with the fix I just added

File: lib/fakes/appeal_repository.rb [hunk 3]
```diff
@@ -284,15 +284,15 @@ def self.seed!
...
-        "123C" => Fakes::AppealRepository.appeal_ready_to_certify,
-        "456C" => Fakes::AppealRepository.appeal_mismatched_docs,
-        "789C" => Fakes::AppealRepository.appeal_already_certified,
-        "321C" => Fakes::AppealRepository.appeal_remand_decided,
-        "654C" => Fakes::AppealRepository.appeal_full_grant_decided,
         "000ERR" => Fakes::AppealRepository.appeal_raises_vbms_error,
         "001ERR" => Fakes::AppealRepository.appeal_missing_data
       }
       50.times.each do |i|
+        records["#{123 + i}C"] = Fakes::AppealRepository.appeal_ready_to_certify
```
Comment (shanear) [inline lib/fakes/appeal_repository.rb]: there's a lot of complexity here. is there a better way to do this?

File: app/views/certifications/confirm.html.erb (hunk 1 above)
Comment (kierachell) [inline app/views/certifications/confirm.html.erb]: This is just a QOL change for smoother testing in demo;
form8 seems to save the entered data from previous certification attempt (in demo at least);
so this way we can test a bunch of "clean slate" appeals in a row;
I can transfer this link to `dev/users` - how about that?

File: lib/fakes/appeal_repository.rb (hunk 3 above)
Comment (kierachell) [inline lib/fakes/appeal_repository.rb]: This is mainly to generate more than one `ready to certify`: form8 seems to preload previous session's data when we "recertify" the appeals;
It starts at 123 because all the tests assume `123C` as the first available record id;

File: lib/fakes/bgs_service.rb (hunk 2 above)
Comment (kierachell) [inline lib/fakes/bgs_service.rb]: Yeah, I changed it before all the other changes came through;
We're going to need a few different deterministic cases for existing EPs; so this will require more work;

File: app/views/certifications/confirm.html.erb (hunk 1 above)
Comment (joofsh) [inline app/views/certifications/confirm.html.erb]: yea, let's move this to dev/users. We shouldn't aim to isolate test-only code as much as possible

File: lib/fakes/appeal_repository.rb (hunk 3 above)
Comment (joofsh) [inline lib/fakes/appeal_repository.rb]: yea this is way too much complexity. This file is going to quickly get out of control in terms of maintaining the tests and dev'ing against expected data. 

If you're specifically trying to test 1 more ready to certify appeal, can we simply add that for now?

File: app/views/certifications/already_certified.html.erb [hunk 4]
```diff
@@ -15,8 +15,8 @@
     being re-certified to the board, Caseflow is not currently able to
     process remand cases.
   </p>
-  <% if current_user.css_id == ENV['TEST_USER_ID'] && params[:id] == ENV["TEST_APPEAL_ID"] %>
-      <%= link_to "Uncertify Appeal", uncertify_certification_path(params[:id]), method: :post %>
+  <% if current_user.css_id == ENV['TEST_USER_ID'] && params[:vacols_id] == ENV["TEST_APPEAL_ID"] %>
```
Comment (joofsh) [inline app/views/certifications/already_certified.html.erb]: Can we move this to the `/dev/users` page?

File: spec/services/dispatch_service_spec.rb [hunk 5]
```diff
@@ -98,7 +98,7 @@
 
     it "does nothing when valid" do
       expect(dispatch.claim.valid?).to be_truthy
-      expect { subject }.to_not raise_error(Dispatch::InvalidClaimError)
+      expect { subject }.to_not raise_error
```
Comment (joofsh) [inline spec/services/dispatch_service_spec.rb]: why this change? The original test was more descriptive

File: lib/fakes/bgs_service.rb (hunk 2 above)
Comment (joofsh) [inline lib/fakes/bgs_service.rb]: Looks like this is just a copy of the original array? You can just do `ep_data = END_PRODUCTS.dup`

File: lib/fakes/bgs_service.rb [hunk 6]
```diff
@@ -24,96 +24,21 @@ class Fakes::BGSService
...
+    rand(0..10).times.each do |num|
```
Comment (joofsh) [inline lib/fakes/bgs_service.rb]: You can just do `rand(X)` so rand(10) should be equivalent

File: lib/fakes/initializer.rb [hunk 7]
```diff
@@ -6,13 +6,14 @@ def self.development!
...
-      "name" => "Anne Merica"
-    }
+    # User.authentication_service.user_session = {
+    #  "id" => "ANNE MERICA",
+    #  "roles" => ["Certify Appeal", "Establish Claim", "Manage Claim Establishment"],
+    #  "station_id" => "283",
+    #  "email" => "america@example.com",
+    #  "name" => "Anne Merica"
+    # }
+    User.authentication_service.user_session = User.find_by(id: "Cave Johnson")
```
Comment (joofsh) [inline lib/fakes/initializer.rb]: Good change overall, but I don't think this will work as it stands? Previously `user_session` was returning a hash, now you're returning a user object. 

Also can we remove the old comment please

Comment (joofsh) [inline lib/fakes/initializer.rb]: Good change overall, but I don't think this will work as it stands? Previously `user_session` was returning a hash, now you're returning a user object. 

Also can we remove the old comment please

Comment (joofsh) [inline lib/fakes/initializer.rb]: Also I see you make two users with the full_name "Cave Johnson", but that wouldn't be their primary `id` here. Looks like this would find no user

File: app/views/certifications/already_certified.html.erb (hunk 4 above)
Comment (kierachell) [inline app/views/certifications/already_certified.html.erb]: I assume `dev/users` page is not available in `UAT`;
This is added here in case the `test appeal` was not "uncertified" during the last testing round. Is there a better way to guarantee that all data is cleaned for each `UAT` test?

File: spec/services/dispatch_service_spec.rb (hunk 5 above)
Comment (kierachell) [inline spec/services/dispatch_service_spec.rb]: Specifying `ErrorType` will allow any other error to be ignored - so the test will pass. Is that intended?

File: lib/fakes/initializer.rb (hunk 7 above)
Comment (kierachell) [inline lib/fakes/initializer.rb]: It doesn't find it by `name` so 🤷‍♂️

File: lib/fakes/appeal_repository.rb (hunk 3 above)
Comment (kierachell) [inline lib/fakes/appeal_repository.rb]: I do need more than 1 appeal to certify, because, at least at this time, the data for Form8 is being pulled from whatever anyone else entered in any other session. So, for example, @nikitarockz will certify `123C` with certain values entered then I go and certify `123C` after that (on a different machine/etc) and my Form8 will be filled with whatever she entered. Bug? Feature?
This is my way of just getting a clean slate each time without breaking all the tests ever.

**Comment by kierachell** [comment]
- Adjust the fake User authentication as requested (even though it didn't seem to affect the functionality for some reason)
- Still need a bunch of certifiable appeals
- Removed extra testing endpoints
- The `raise_error` test catches all errors now, not just specific type (the type of error is displayed on fail anyway)

File: spec/services/dispatch_service_spec.rb (hunk 5 above)
Comment (shanear) [inline spec/services/dispatch_service_spec.rb]: > Specifying `ErrorType` will allow any other error to be ignored - so the test will pass. Is that intended?
It's the opposite. If we don't specify the error type, other errors will incorrectly cause the test to pass.

agree with @joofsh here, we should be looking for a specific error.

File: lib/fakes/appeal_repository.rb (hunk 3 above)
Comment (shanear) [inline lib/fakes/appeal_repository.rb]: @kierachell don't we have an /uncertify endpoint that we can use for this purpose?

File: spec/services/dispatch_service_spec.rb (hunk 5 above)
Comment (kierachell) [inline spec/services/dispatch_service_spec.rb]: This is the message I am seeing when `ErrorType` is specificed:
screen shot 2017-01-24 at 14 24 50

Am I reading it wrong?

File: lib/fakes/appeal_repository.rb (hunk 3 above)
Comment (kierachell) [inline lib/fakes/appeal_repository.rb]: The `/uncertify` is gated by test user id, do you want me to set it in `demo` env and specific appeal id - I can set those up instead and add uncertify to `dev/users`, but I was trying to keep that footprint as isolated as possible.
